django/contrib/admin/sites.py

```diff
@@ -245,20 +247,20 @@ def wrapper(*args, **kwargs):
 
         # Add in each model's views, and create a list of valid URLS for the
         # app_index
-        valid_app_labels = []
         for model, model_admin in self._registry.items():
+            base_path = r'%s/%s/' % (model._meta.app_label, model._meta.model_name)
             urlpatterns += [
-                url(r'^%s/%s/' % (model._meta.app_label, model._meta.model_name), include(model_admin.urls)),
+                url(r'^%s' % base_path, include(model_admin.urls)),
+                url(r'^%s(?P<field_name>.+)/$' % base_path,
```

Comment:
Oh, good catch with the add! What url patter would you prefer. I need to put it somewhere and the patterned includes the `app_label` and `model_name` now due to some security concerns from @apollo13 .